From 4e94fde4bc72f3dbcd0978a8c7f19f4cc56d7ebe Mon Sep 17 00:00:00 2001
From: Andre Nahrwold <andre.nahrwold@wago.com>
Date: Mon, 11 Sep 2017 16:57:16 +0200
Subject: [PATCH] dts: pfc8215: activate usb0 as host interface

1) dts: pfc8215: activate and configure tps ls2 for usb power
2) dts: pfc8215: delete activating usb1
3) dts: pfc8215: activate usb0 as host interface
---
 arch/arm/boot/dts/wago-pfc200v3-devconf-0182.dts | 36 ++++++++++++++++++++----
 1 file changed, 31 insertions(+), 5 deletions(-)

diff --git a/arch/arm/boot/dts/wago-pfc200v3-devconf-0182.dts b/arch/arm/boot/dts/wago-pfc200v3-devconf-0182.dts
index 9236833..96df66b 100644
--- a/arch/arm/boot/dts/wago-pfc200v3-devconf-0182.dts
+++ b/arch/arm/boot/dts/wago-pfc200v3-devconf-0182.dts
@@ -30,7 +30,6 @@
 		usv_irq = <&gpio1 17 GPIO_ACTIVE_LOW>;
 		switch-reset = <&gpio3 10 GPIO_ACTIVE_LOW>;
 		kbus-reset =  <&gpio1 25 GPIO_ACTIVE_HIGH>;
-		TPS65218-LS2 = "LS2_off"; /* switch USB off */
 		led-off;
 	};
 
@@ -66,12 +65,39 @@
 	};
 };
 
-&usb1_phy {
-	status = "okay";
+&i2c0 {
+        tps65218@24 {   /* Power Managment IC  */
+                loadswitch2:  loadswitch-ls2 {
+                        compatible = "ti,tps65218-ls2";
+                        /* VDD_LS2 +5V15 for USB */
+                        regulator-name = "vdd_LS2_USB";
+                        regulator-min-microamp = <200000>; /* Should be set to 100 mA according to SLP00000843.002 */
+                        regulator-max-microamp = <200000>; /* but 4G Modem is not being inumurated correctly, so we increase it */
+                        regulator-boot-on;
+                        regulator-always-on;
+                };
+        };
 };
 
-&usb1 {
-	status = "okay";
+&usb {
+        status = "okay";
+};
+
+&usb_ctrl_mod {
+        status = "okay";
+};
+
+&usb0_phy {
+        status = "okay";
+};
+
+&usb0 {
+        status = "okay";
+        dr_mode = "host";
+};
+
+&cppi41dma {
+        status = "okay";
 };
 
 &ksz8863 {
-- 
2.7.4

